src/Lib/Managers/NobManager.vala

```diff
@@ -103,21 +104,18 @@ public class Akira.Lib.Managers.NobManager : Object {
         this.selected_nob = selected_nob;
     }
 
-    /**
-     * Compares a target item to the current nobs to see if there is a match.
-     * Otherwise returns Nob.NONE.
-     */
-    public Nob get_grabbed_id (Goo.CanvasItem? target) {
-        int grabbed_id = Nob.NONE;
-
-        for (var i = 0; i < 9; ++i) {
-            if (target == nobs[i]) {
-                grabbed_id = i;
-                break;
+    public Nob hit_test (double x, double y) {
+        double scale = canvas.current_scale;
+        foreach (var ui_nob_s in nobs) {
+            if (ui_nob_s != null && ui_nob_s.is_visible ()) {
+                var ui_nob = ui_nob_s as Akira.Lib.Selection.Nob;
+                if (ui_nob.hit_test (x, y, scale)) {
+                    return ui_nob.handle_id;
+                }
             }
         }
 
-        return (Nob) grabbed_id;
+        return Nob.NONE;
```

Comment:
I think we could update the attribute to be an array of Selection.Nob instead of Goo.CanvasItemSimple.
`private Lib.Selection.Nob[] nobs = new Lib.Selection.Nob[9];`
Not sure if that affects performance, but in that way we can avoid type casting these nobs every time and simplify the code.

```
foreach (var nob in nobs) {
            if (nob == null || !nob.is_visible ()) {
                continue;
            }

            if (nob.hit_test (x, y, scale)) {
                return nob.handle_id;
            }
}
```

Reply:
changed it.